web/static/app/src/controllers/history_controller.js

```
@@ -73,14 +78,32 @@ export default class extends Controller {
     }
   }
 
+  changeTransactionType () {
+    this.historyTableTarget.innerHTML = ''
+    this.transactionCountTarget.setAttribute('data-tx-count', 0)
+    hide(this.transactionCountContainerTarget)
+    this.nextPage = 1
+    this.fetchMoreTxs()
+  }
+
   fetchMoreTxs () {
     show(this.loadingIndicatorTarget)
 
+    const txType = this.transactionTypeTarget.value
+
     const _this = this
-    axios.get(`/next-history-page?page=${this.nextPage}`)
+    axios.get(`/next-history-page?page=${this.nextPage}&tx-type=${txType}`)
       .then(function (response) {
         let result = response.data
         if (result.success) {
+          if (!result.txs || result.txs.length === 0) {
```

`result.success` is true but there's either no txs in result or txs count is 0? This shouldn't happen. Why are we writing code for such possibility?

If `result.success` is true, then txs should be returned. Unless there's no tx for the type specified (in which case it shouldn't be on the filter) or the page used in the query does not have data to return (in which case the handler should return an error message, not `result.success = true`; the error would be that there's `no transaction for page  ${x}`